IntegrationTests/Services/AWSKinesisIntegrationTests/KinesisTests.swift

```diff
@@ -0,0 +1,104 @@
+            guard case .success(let stream) = describeStreamOutput.result else { throw KinesisTestError("could not describe stream") }
+            let streamARN = stream.streamDescription?.streamARN
+
+            // Make a set of 10 records, add them to the stream
+            var recordStrings = (1...10).map { _ in UUID().uuidString }
+            for record in recordStrings {
+                let putRecordInput = PutRecordInput(data: record.data(using: .utf8), explicitHashKey: nil, partitionKey: "Test", sequenceNumberForOrdering: nil, streamName: streamName)
+                let _ = try await client.putRecord(input: putRecordInput)
+            }
+
+            // Get the first (only) shard in the data stream
+            let listShardsInput = ListShardsInput(streamName: streamName)
+            let shardList = try await client.listShards(input: listShardsInput)
+            let shard = shardList.shards?.first!
+
+            // Create a consumer for the shard
+            let consumerName = UUID().uuidString
+            let consumerInput = RegisterStreamConsumerInput(consumerName: consumerName, streamARN: stream.streamDescription?.streamARN)
+            let consumer = try await client.registerStreamConsumer(input: consumerInput)
+            let consumerARN = consumer.consumer?.consumerARN
+
+            // Wait until the consumer becomes active
+            let describeConsumerInput = DescribeStreamConsumerInput(consumerARN: consumerARN, streamARN: streamARN)
+            var counter = 30
+            repeat {
+                try await Task<Never, Never>.sleep(nanoseconds: 1_000_000_000)
+                let out = try await client.describeStreamConsumer(input: describeConsumerInput)
+                if out.consumerDescription?.consumerStatus == .active { break }
+                counter -= 1
+                if counter < 0 { throw KinesisTestError("consumer timed out while waiting for active") }
+            } while true
+
+            // Create the subscription stream
+            let input = SubscribeToShardInput(consumerARN: consumerARN, shardId: shard?.shardId, startingPosition: KinesisClientTypes.StartingPosition(sequenceNumber: shard?.sequenceNumberRange?.startingSequenceNumber, type: .atSequenceNumber))
+            let output = try await client.subscribeToShard(input: input)
+
+            // Monitor the shard event stream
+            for try await event in output.eventStream! {
+                switch event {
+                case .subscribetoshardevent(let event):
+                    event.records?.forEach { record in
+                        let recordString = String(data: record.data ?? Data(), encoding: .utf8)
+                        recordStrings.removeAll { recordString == $0 }
+                    }
+                case .sdkUnknown(let message):
+                    print("Unknown event: \(message)")
```

Comment:
The first event to come down the stream is of type `initial-response`:

```
2023-04-28T16:29:53-0500 info AWSMessageDecoder : [Logging] onHeaderReceived: :event-type: string(value: "initial-response")
```

This event is caught & printed out here as unknown.

```
Unknown event: error processing event stream, unrecognized event: initial-response
```

This is followed by an event of type `SubscribeToShardEvent` which is defined in the operation.

Reply:
This is actually a feature defined in Smithy:
https://smithy.io/2.0/spec/streaming.html?highlight=event%20stream#initial-messages

We need to determine if initial messages should be handled by the SDK versus passed down the event stream to the consumer.

Reply:
GH issue for tracking handling of initial events: https://github.com/awslabs/aws-sdk-swift/issues/976